pkg/kv/kvserver/allocator/base.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package allocator
+
+import (
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/constraint"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+)
+
+// MaxCapacityCheck returns true if the store has room for a new replica.
+func MaxCapacityCheck(store roachpb.StoreDescriptor) bool {
+	return store.Capacity.FractionUsed() < MaxFractionUsedThreshold
+}
+
+const (
+	// MaxFractionUsedThreshold controls the point at which the store cedes having
+	// room for new replicas. If the fraction used of a store descriptor capacity
+	// is greater than this value, it will never be used as a rebalance or
+	// allocate target and we will actively try to move replicas off of it.
+	MaxFractionUsedThreshold = 0.95
+)
```

Comment:
Why did these move into a separate file? There are other similar constraint checks that should probably also tag along if these do get split here.

nit: the const declaration should be above the func.

Reply:
Everything that's shared between the subpackages found its way below into this base package, everything that could be scoped to just the subpackage was kept within. MaxCapacityCheck is used by both the allocatorimpl and storepool, which is why only these constants feature here. These boundaries are going to be fuzzy/awkward for now, it's not clear what things should be shared vs. not.

Moved the const declarations above.